src/include/btree_cmp_inline.h

```diff
@@ -14,13 +14,158 @@
 #endif
 #endif
 
-#if defined(HAVE_ARM_NEON_INTRIN_H)
-#include <arm_neon.h>
-#endif
-/* 16B alignment */
-#define WT_ALIGNED_16(p) (((uintptr_t)(p)&0x0f) == 0)
 #define WT_VECTOR_SIZE 16 /* chunk size */
 
+#ifndef HAVE_X86INTRIN_H
+/*
+ * __lex_compare_ge_16 --
+ *     Lexicographic comparison routine for data greater than or equal to 16 bytes. Returns: < 0 if
+ *     user_item is lexicographically < tree_item, = 0 if user_item is lexicographically =
+ *     tree_item, > 0 if user_item is lexicographically > tree_item. We use the names "user" and
+ *     "tree" so it's clear in the btree code which the application is looking at when we call its
+ *     comparison function. Some platforms like ARM offer dedicated instructions for reading 16
+ *     bytes at a time, allowing for faster comparisons.
+ */
+static WT_INLINE int
+__lex_compare_ge_16(const uint8_t *ustartp, const uint8_t *tstartp, size_t len, int lencmp)
+{
+    struct {
+        uint64_t a, b;
+    } udata, tdata;
+    uint64_t u64, t64;
+    const uint8_t *uendp, *userp, *tendp, *treep;
+    bool firsteq;
+
+    uendp = ustartp + len;
+    tendp = tstartp + len;
+
+    /* skip 16 matching bytes at a time, starting at first possible difference. */
+    for (userp = ustartp, treep = tstartp; uendp - userp > WT_VECTOR_SIZE;
+         userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
+        memcpy(&udata, userp, WT_VECTOR_SIZE);
+        memcpy(&tdata, treep, WT_VECTOR_SIZE);
+        if (udata.a != tdata.a || udata.b != tdata.b)
+            goto final128;
+    }
+
+    /*
+     * Rewind until there is exactly 16 bytes left. We know we started with at least 16, so we are
+     * still in bound.
+     */
+    memcpy(&udata, uendp - WT_VECTOR_SIZE, WT_VECTOR_SIZE);
+    memcpy(&tdata, tendp - WT_VECTOR_SIZE, WT_VECTOR_SIZE);
+
+final128:
+    firsteq = udata.a == tdata.a;
+    u64 = firsteq ? udata.b : udata.a;
+    t64 = firsteq ? tdata.b : tdata.a;
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    u64 = __builtin_bswap64(u64);
```

Comment:
Change all occurrences to
```suggestion
    u64 = __wt_bswap64(u64);
```
please.